### ISSUE TITLE
[Bug]: Timetable speed is set to input-1 (not for all values)

### ISSUE THREAD

**Issue by Transportman** [OpenTTD/OpenTTD#12891]
### Version of OpenTTD

14.1, Windows 10

### Expected result

I expect that the speed limit I put in, is also the speed limit set.

### Actual result

Some values are shown as input-1 for the speed limit. I tested with various values, and +-90 to +-170 km/h and +-260 to +-340 km/h give the described behaviour.

### Steps to reproduce

1. Build a train
2. Give it some orders
3. Open the timetable window
4. Set the speed limit to 100 km/h
5. Speed limit is set to 99 km/h instead of 100 km/h
6. Set the speed limit to 101 km/h
7. Speed limit is now set to 100 km/h
8. Change speed limit and just OK the window without changing the value
9. Speed limit stays 100 km/h

This is not present in 13.x, there the 100 km/h is correctly shown as 100 km/h on the first input.

**Comment by abi9ail** [comment]
**TL;DR: Rounding**

This behaviour is due to rounding errors when converting between displayed units and the "km/h-ish" maximum order speed. 
The entered value is [converted](https://github.com/OpenTTD/OpenTTD/blob/7092162a97e3b31e17f30fad11ee1b0106bd5f65/src/timetable_gui.cpp#L743) to "km/h-ish" by dividing the value by its unit's [conversion factor to internal speed](https://github.com/OpenTTD/OpenTTD/blob/7092162a97e3b31e17f30fad11ee1b0106bd5f65/src/strings.cpp#L856) and then multiplying by 1.6. This process is performed in reverse when displaying an order's maximum speed.

Whilst some changes to rounding can resolve this issue for other display units, I don't believe that km/h conversion can be resolved without some kind of change to the underlying type. As the entered value is divided by a factor greater than 1 when using km/h as display units, and the returned value is an integer, the best case scenario is that only a few* entered values cause `ConvertDisplaySpeedToDisplaySpeed` to return the same "km/h-ish" value.

Happy to have a look into writing a fix but I'd prefer to know what the preferred approach to this would be.

$\text{*Approx every}\frac{\frac{16}{10}}{1.609344-\frac{16}{10}}\approx\text{171 km/h.}$
